common/rya.api/src/main/java/org/apache/rya/api/client/ExecuteSparqlQuery.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.api.client;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Loads a SPARQL Query and executes the query against an instance of Rya.
+ */
+@DefaultAnnotation(NonNull.class)
+public interface ExecuteSparqlQuery {
+    /**
+     * Loads a SPARQL Query and executes the query against an instance of Rya.
+     *
+     * @param ryaInstanceName - The name of the Rya instance the query will be executed against. (not null)
+     * @param sparqlQuery - A single SPARQL Query. (not null)
+     * @return A user-friendly String representation of the query results.
```

Comment:
Yup.  I thought about that.  Current implementation is limited to a string of length Integer.MAX_INT.  That is a more advanced feature that I think is beyond the scope of this PR.  Besides, you can always use LIMIT and OFFSET in your sparql query to get around this limitation.  I've been adding out of scope features as I go to this https://issues.apache.org/jira/browse/RYA-314 as subtasks.